### PR TITLE
Add cume_dist Presto window function

### PR DESCRIPTION
**Pull request by pramodsatya**

### CODE REVIEW

File: velox/functions/prestosql/window/tests/CumeDistTest.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/tests/CumeDistTest.cpp:22]: You could combine these tests with RankTest as this function too should always go along all of them.

File: velox/functions/prestosql/window/CumeDist.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/CumeDist.cpp:44]: Nit -- Rename variables to peerGroupStartsVector and peer GroupEndsVector.

File: velox/functions/prestosql/window/CumeDist.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/CumeDist.cpp:47]: Nit -- Maybe peerStart or peerGroupStart might be more consistent name.

**Review by aditi-pandit — COMMENTED** [review]
Thanks Pramod for this function.

File: velox/functions/prestosql/window/tests/CumeDistTest.cpp
Comment (pramodsatya) [reply velox/functions/prestosql/window/tests/CumeDistTest.cpp:22]: Thanks for the suggestion, that is what I wanted to do initially actually, I will combine these!

**Comment by aditi-pandit** [comment]
Please rebase after https://github.com/facebookincubator/velox/pull/2700 lands

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (pramodsatya) [reply velox/functions/prestosql/window/tests/RankTest.cpp]: Thanks for the suggestion, changed accordingly!

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/tests/RankTest.cpp:44]: Can you initialize here itself and make it a const vector ?

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/tests/RankTest.cpp:31]: Make this const auto&

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (aditi-pandit) [inline velox/functions/prestosql/window/tests/RankTest.cpp:39]: Make this const auto& also

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (mbasmanova) [inline velox/functions/prestosql/window/tests/RankTest.cpp:26]: const &

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (mbasmanova) [inline velox/functions/prestosql/window/tests/RankTest.cpp:40]: Consider turning this test into parameterized test using function name as a parameter.

https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests

**Review by mbasmanova — COMMENTED** [review]
@aditi-pandit Looks good to me % a couple of small comments.

**Comment by aditi-pandit** [comment]
> @aditi-pandit Looks good to me % a couple of small comments.

@pramodsatya is the author of this PR. Sorry for the confusion. I only started your review after mine.

File: velox/functions/prestosql/window/tests/RankTest.cpp
Comment (pramodsatya) [reply velox/functions/prestosql/window/tests/RankTest.cpp]: Thanks for the suggestion, modified accordingly!

**Review by mbasmanova — APPROVED** [review]
@pramodsatya Thanks. I think you need to wait for #2702 to land, then rebase. Ping me after that so I can merge.

**Comment by pramodsatya** [comment]
@mbasmanova squashed commits and rebased on top of #2702 changes, thank you!

**Comment by aditi-pandit** [comment]
https://github.com/facebookincubator/velox/issues/685

**Pull request closed without merging**
